utilities/cheyenne/launch-dask.py

```diff
@@ -0,0 +1,131 @@
+
+    search_jobid = re.search("^(\S+)$", output.decode('utf-8'))
+    jobid = search_jobid.group(1)
+    return jobid
+
+def start_jlab(dask_scheduler, host=None, port='8888', notebook_dir=''):
+    cmd = ['jupyter', 'lab', '--ip', host,
+           '--no-browser', '--port', port,
+           '--notebook-dir', notebook_dir]
+#    logger.debug("cmd={}".format(cmd))
+    proc = subprocess.Popen(cmd)
+    dask_scheduler.jlab_proc = proc
+
+def get_logger(log_level):
+    ch = logging.StreamHandler(sys.stdout)
+    formatter = logging.Formatter(' - '.join(
+        ["%(asctime)s", "%(name)s", "%(levelname)s", "%(message)s"]))
+    ch.setFormatter(formatter)
+    logger = logging.getLogger(__file__)
+    logger.setLevel(log_level)
+    ch.setLevel(log_level)
+    logger.addHandler(ch)
+    return logger
+
+
+
+def setup_jlab(scheduler_file, jlab_port, dash_port, notebook_dir,
+        hostname):
+
+    logger.info('getting client with scheduler file: %s' % scheduler_file)
+    client = Client(scheduler_file=scheduler_file, timeout=30)
+    logger.debug('Client: %s' % client)
+
+    logger.debug('Getting hostname where scheduler is running')
+    host = client.run_on_scheduler(socket.gethostname)
+    logger.info('host is %s' % host)
+
+    logger.info('Starting jupyter lab on host')
+    client.run_on_scheduler(start_jlab, host=host, port=jlab_port,
+                            notebook_dir=notebook_dir)
+    logger.debug('Done.')
+
+    user = os.environ['USER']
+    print('Run the following command from your local machine:')
+    print(f'ssh -N -L {jlab_port}:{host}:{jlab_port} '
+          f'-L {dash_port}:{host}:8787 {user}@{hostname}')
+    print('Then open the following URLs:')
+    print(f'\tJupyter lab: http://localhost:{jlab_port}')
+    print(f'\tDask dashboard: http://localhost:{dash_port}', flush=True)
+
+def _main_func(args, description):
+    project, workers, walltime, notebookdir, workdir = parse_command_line(args, description)
+    global logger
+    logger = get_logger("DEBUG")
+
+    jobids = []
+    jobids.append( launch_job("launch-dask-scheduler.sh", project, walltime))
+
+    for i in range(workers):
+        jobids.append(launch_job("launch-dask-worker.sh", project, walltime))
+
+    for jobid in jobids:
+        wait = True
+        while wait:
+            proc = subprocess.Popen("qstat {} ".
+                                    format(jobid),
+                                    stdout=subprocess.PIPE,stderr=subprocess.PIPE,
+                                    shell=True)
+
+            output, errput = proc.communicate()
+            output = output.decode('utf-8')
+            stat = proc.wait()
+            if ' R ' in output:
+                logger.info(" jobid {} started".format(jobid))
+                time.sleep(5) # make sure scheduler file is written
+                wait = False
+            if ' Q ' in output:
+                logger.info(" jobid {} in queue".format(jobid))
+                time.sleep(5)
```

Comment:
We only need to block until the scheduler job is running. The rest of the worker jobs can come later.